warehouse/integrations/deltalake-native/deltalake_native_test.go

```diff
@@ -0,0 +1,388 @@
+package deltalake_native_test
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"testing"
+
+	"github.com/rudderlabs/rudder-server/warehouse/integrations/deltalake"
+
+	deltalake_native "github.com/rudderlabs/rudder-server/warehouse/integrations/deltalake-native"
+
+	"github.com/rudderlabs/rudder-server/warehouse/encoding"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/rudderlabs/rudder-server/warehouse/validations"
+
+	"github.com/rudderlabs/rudder-server/utils/misc"
+
+	backendconfig "github.com/rudderlabs/rudder-server/backend-config"
+
+	warehouseclient "github.com/rudderlabs/rudder-server/warehouse/client"
+	warehouseutils "github.com/rudderlabs/rudder-server/warehouse/utils"
+	"github.com/stretchr/testify/require"
+
+	"github.com/rudderlabs/rudder-server/warehouse/integrations/testhelper"
+)
+
+func TestIntegrationDeltalake(t *testing.T) {
+	if os.Getenv("SLOW") == "0" {
+		t.Skip("Skipping tests. Remove 'SLOW=0' env var to run them.")
+	}
+	if _, exists := os.LookupEnv(testhelper.DeltalakeIntegrationTestCredentials); !exists {
+		t.Skipf("Skipping %s as %s is not set", t.Name(), testhelper.DeltalakeIntegrationTestCredentials)
+	}
+
+	t.Parallel()
```

Comment:
You might get data races on `pkgLogger` here. Perhaps it is time to remove `pkgLogger` after all :slightly_smiling_face:

Reply:
https://github.com/rudderlabs/rudder-server/pull/3202